### ISSUE TITLE
Invalid pixel intensity values for rescaling with non-linear interpolation

### ISSUE THREAD

**Issue by grigorisg9gr** [menpo/menpo#747]
When rescaling an image and selecting a higher order of 
interpolation, i.e a non-linear one, it is likely that some values
will be out of the [0, 1] valid interval. 

Code sample: 
```
im = mio.import_builtin_asset.lenna_png()
im1 = im.rescale_landmarks_to_diagonal_range(120, order=2)
px = im1.pixels
print('{:.03f}, {:.03f}'.format(np.min(px), np.max(px)))
```

I am not sure if you would like to do anything about it, but either a warning or
a value clipping would make sense. I could go ahead with the clipping if you
would like it like that.

**Comment by patricksnape** [comment]
Yes we could provide the ability to auto-clip? But how far would that propagate - I'm reluctant if every single transformation function needs this clipping... Maybe we could add a method that will clip pixels values to images?

**Comment by grigorisg9gr** [comment]
I agree that would be a computational burden. One idea consistent with the image class is to have an optional argument for the rescale_*() (similar to warp_landmarks=True), which if True, only then clipping would be performed.

**Comment by patricksnape** [comment]
It's not the computational burden - it's that every function that does warping needs the new `kwarg` for clipping which is a lot of changes and a lot of places to forget. Which we can still do it's just I'm not sure how I feel about it. Much like how `order` is already propagated all over the place which is a pain.
